src/to-have-form-values.js

```diff
@@ -69,9 +63,15 @@ export function toHaveFormValues(formElement, expectedValues) {
   }
   const formValues = getAllFormValues(formElement)
   return {
-    pass: Object.entries(expectedValues).every(([name, expectedValue]) =>
-      isEqualWith(formValues[name], expectedValue, compareArraysAsSet),
-    ),
+    pass: Object.entries(expectedValues).every(([name, expectedValue]) => {
+      if (Array.isArray(formValues[name]) && Array.isArray(expectedValue)) {
+        return [...new Set(formValues[name])].every(v =>
+          new Set(expectedValue).has(v),
+        )
+      } else {
+        return formValues[name] === expectedValue
+      }
+    }),
```

Comment:
The one thing lost here is readability. It is easier to read the previous version.

What I suggest is that you create a `function isEqualWith` (or `isEqualWithArraysAsSets`) in the `utils.js` module, that would encapsulate this functionality. Makes sense?